101-machine-learning-create/azuredeploy.json

```diff
@@ -2,257 +2,641 @@
+    {
+      "condition": "[or(not(equals(parameters('vnetOption'), 'none')), not(equals(parameters('privateEndpointType'), 'none')))]",
+      "type": "Microsoft.Network/virtualNetworks",
+      "apiVersion": "2019-09-01",
+      "name": "[variables('vnetName')]",
+      "location": "[if(variables('isLocationValid'), parameters('location'), 'InvalidLocation')]",
+      "properties": {
+        "addressSpace": {
+          "addressPrefixes": "[parameters('addressPrefixes')]"
+        },
+        "enableDdosProtection": false,
+        "enableVmProtection": false
+      }
+    },
+    {
+      "condition": "[or(not(equals(parameters('subnetOption'), 'none')), not(equals(parameters('privateEndpointType'), 'none')))]",
+      "type": "Microsoft.Network/virtualNetworks/subnets",
+      "apiVersion": "2019-09-01",
+      "name": "[concat(variables('vnetName'), '/', variables('subnetName'))]",
+      "dependsOn": [
+        "[variables('vnet')]"
+      ],
+      "properties": {
+        "addressPrefix": "[parameters('subnetPrefix')]",
+        "privateEndpointNetworkPolicies": "Disabled",
+        "privateLinkServiceNetworkPolicies": "Enabled",
+        "serviceEndpoints": [
+          {
+            "service": "Microsoft.Storage"
+          },
+          {
+            "service": "Microsoft.KeyVault"
+          },
+          {
+            "service": "Microsoft.ContainerRegistry"
+          }
         ]
+      }
+    },
+    {
+      "condition": "[equals(parameters('storageAccountOption'), 'new')]",
+      "type": "Microsoft.Storage/storageAccounts",
+      "apiVersion": "2019-04-01",
+      "name": "[variables('storageAccountName')]",
+      "location": "[if(variables('isLocationValid'), parameters('location'), 'InvalidLocation')]",
+      "sku": {
+        "name": "[parameters('storageAccountType')]"
       },
-      "storageAccountResourceGroupName": {
-        "type": "string",
-        "defaultValue": "[resourceGroup().name]"
-      },
-      "keyVaultNewOrExisting": {
-        "type": "string",
-        "defaultValue": "new",
-        "allowedValues": [
-          "new",
-          "existing"
-        ],
-        "metadata": {
-          "description": "Determines whether or not a key vault should be provisioned."
+      "kind": "StorageV2",
+      "properties": {
+        "encryption": {
+          "services": {
+            "blob": {
+              "enabled": true
+              },
+            "file": {
+              "enabled": true
+            }
+          },
+          "keySource": "Microsoft.Storage"
+        },
+        "supportsHttpsTrafficOnly": true
+    }
+    },
+    {
+      "condition": "[equals(parameters('storageAccountBehindVNet'), 'true')]",
+      "type": "Microsoft.Resources/deployments",
+      "apiVersion": "2017-05-10",
+      "name": "UpdateStorageAcls",
+      "dependsOn": [
+          "[variables('subnet')]",
+          "[variables('storageAccount')]"
+      ],
+      "resourceGroup": "[resourceGroup().name]",
```

Comment:
What's the intent here? is it to update an existing storage account?  If so, this needs to be set to the resourceGroup for that storage account, not the current RG.

Reply:
@bmoore-msft Yes, update the ACL for new or existing storage account if behind Vnet